app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
```

Comment:
Well, you're right that we do have some repetition in the forms right now. And I agree we should get rid of it.

**But**.. this is a components that just does buttons. That's not the place to be doing redirects, nor is it a place to change the parent form's data.

We can always put the common code in a `FormController` and inherit all the `*FormControllers` from that one. Or have a `miqService` function that injects the default implementations of those funcions on the controller object.

Having smart buttons is simply a layering violation.

---

Besides, have you actually looked at `cancel` across different forms? Some redirect you to a specific url, some just do `miqAjaxButton` and expect the server to redirect you... The simple fact that there is not just one `cancelClicked` implementation means this makes the component much less usable.